### PR TITLE
Create CuratedContentEditorialWorkflow.md

### PR DESCRIPTION
**Pull request by markcmiller86**
These things need further consideration:

This PR is based on the original draft from @bernhold [Google doc](https://docs.google.com/document/d/19PYkAKMunpJAzchfKEhdBrjtq0X_lf69cDNz3iTXfzU/edit#).

We are submitting here to engage in discussion using conversation in this PR and iteration to develop a consensus on workflow for curated content.

The goal is to keep the focus on curated content workflow for time being.

### CODE REVIEW

**Comment by markcmiller86** [comment]
From @bernhold 

*   This workflow is different from (and much simpler than) the [workflow for blog articles](https://docs.google.com/document/d/1j1-j3Viuu6urxRmlvViiwm-UlLXQ-NJMk3U7a68rk3s/edit?usp=sharing)
*   I think, to make this work, we’re going to have to create an Editorial Board “team” in the betterscientificsoftware repository and limit write access to the repo to them.  
    *   So far, we’ve been giving direct write access to just about anyone.
    *   There are others in the blog workflow who may need to have write access (or maybe should just work by PRs).
*   Speculation: this workflow might also be appropriate for non-blog original articles, but we might want a little more rigor in that type of content.  It is probably too heavyweight for Events.

**Comment by markcmiller86** [comment]
*   Perhaps part of the issue submission should include some assessment of the depth of review of the material upon which the suggestion is based perhaps by assigning one of these labels...
    *   Critical read: cover to cover, critical read of the material
        *   In which case, the submission should include some commentary supporting why the suggested content is relevant/good for BSSw
    *   Cursory read: abstract, results, targeted paragraphs/figures
    *   Coincidental: Did not read. Just happened to run across this and thought it could be useful.
    *   Note use of “read” in above works only for written content. We have videos and podcasts too. So, maybe replace “read” with “evaluation”
*   It is a tad heavy-weight process for small things. I think maybe the criteria for when the process is appropriate is maybe length in word count or when the content shares evaluative (qualitative or quantitative) comments, opinions, experiences, etc. then this process is needed? That would probably not include events ;)
    *   db: I didn’t envision using this for Events.  But I think if we were able to automate many of the steps, it wouldn’t actually need to be that hard or heavy-weight.  But I’m open to other opinions
*   I like the idea of automation as much as possible but that shouldn’t be a deal breaker for implementation...I think manual labeling processes can mitigate any lack of automation.
*   Step 4 seem like its a vote-based and a bit different from what was discussed in last week’s telecon where we could move forward if at least two *independent* favorable judgements were made. Who votes? Just EBMs (makes sense to me)? How close a look at the content is required before one’s vote can be legitimately cast?
    *   db: Yes, it is a bit different.  In thinking how we might be able to automate, I realized that a contentious item might have a lot of discussion and various people weigh in with +1 or -1 votes.  It seemed to me that looking for an excess of positive votes was feasible from an automation standpoint and hopefully, more or less equivalent to what we discussed.
    *   db: I envisioned this as EB members votes counting.  We can define as we like, but I think for automation, we’d have to create them as teams in GH or something “official”.
    *   db: I’m assuming people are responsible enough to take an appropriate look at the proposed content before engaging in discussion or voting.  If not, we probably don’t want them on the EB.
*   Regarding 4b...you can search issues by open date, [https://help.github.com/en/articles/searching-issues-and-pull-requests#search-by-open-or-closed-state](https://help.github.com/en/articles/searching-issues-and-pull-requests#search-by-open-or-closed-state). I am thinking an issue that goes in month before SC conference might not get attention more because people are busy than because the issue doesn’t have merit.
*   Step 5 seems like its on a volunteer basis. Maybe it would be good for a deadline and candidate author(s) to be identified, perhaps by using the *Assignees* feature in GitHub. We could have an “Author Pool” and those who agree to be in the pool agree to being assigned, judiciously, this work.
    *   db: Maybe the default assignment is to the person who submitted the issue?
*   I really don’t think the Kanban board is necessary to handle this workflow. I think it could be handled all by labeling on the issues as they move through various phases. In that case, the Kanban board is replaced by pre-canned issue search criteria (which GitHub allows you to do). Conversation in issues is sent to whole team and to submitter of issue (which may be a blessing or a curse). I don’t think same is true for Kanban activity, is it?  A label-only approach is a bit lighter weight too.
    *   db: One of my concerns about using GH tools and processes for all BSSw.io content development is that I think different types of content need different workflows, and if we reach the levels of activity we strive for, we’re going to have a LOT of traffic in the issues area.  My thought was that having multiple project boards would be useful to help us all visualize and follow the different processes for the different types of content.  Also, the project boards support some automations which may (or may not) be useful to us.  I don’t think you can do the same on bare issues.
*   Love the idea of doing some “CI” checks for these PRs...maybe spell checking? Maybe potentially contentious phraseology?

File: Site/CuratedContentEditorialWorkflow.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+# Workflow
+1. **Anyone** creates a GitHub issue for a proposed curated content (CC) item
+   * We should use issue templates to tailor the requested data and automatically
+   apply an appropriate label/project board assignment
+1. The issue is added to the “Curated Content Ideas” (CCI) project board *Idea Backlog* column.
+   * Ideally, this should be done automatically on creation of the issue.  From research on the web,
+     it doesn’t appear that issue templates allow direct assignment to boards yet.  Maybe this can be
+     done with a bot for issues with the appropriate label. *This can be done automatically by a setting
+     on the project board column.*
```
Comment (markcmiller86) [inline Site/CuratedContentEditorialWorkflow.md]: from @bernhold 

Not selectively. We only want curated content ideas to go onto this board. The current automation capabilities only place ALL new issues on a board.

File: Site/CuratedContentEditorialWorkflow.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+1. Discussion and reaction proceeds among the **EB**.  When there are two or more “thumbs up” than
+   “thumbs down” reactions on a topic in review, it moves to the Item *Development Backlog* column.
+   * I think this can be automated via GitHub’s API.
+   * We need to be able to say that if an issue has not reached +2 in a certain time frame, it is
+     considered not appropriate as a CC item.  This probably needs to set a label in the issue
+     indicating this decision, and move it to the _Done_ column.  An example of a rule that might
+     be workable is that there is no further activity on the issue (new discussion or reactions)
+     within 30 days.  There could be a bot that reminds the Editorial Board when issues in
+     Topic Review are aging.
+1. An interested **Author** picks an issue from the *Development Backlog* column and produces a
```
Comment (markcmiller86) [inline Site/CuratedContentEditorialWorkflow.md]: from @prwolfe 

The "add card" dialog let me search for PR's. There are none at the moment so it came back empty but issues and PR's are almost the same in GitHub so this should work.

Comment (markcmiller86) [reply Site/CuratedContentEditorialWorkflow.md]: from @bernhold 

And there are automations on project boards that can be applied to PRs. I just have no experience, so there may well be nuances here that I'm not aware of.

Comment (markcmiller86) [inline Site/CuratedContentEditorialWorkflow.md]: from @rinkug 

Need to define "interested author" and whether original authors will still have the time/inclination to re-work on this

Comment (markcmiller86) [reply Site/CuratedContentEditorialWorkflow.md]: from @bernhold 

In my mind, we give the submitter first crack. If they decline, we have two options: recruit someone, or leave it on the board and wait for some to become interested enough to pick it up.

**Comment by markcmiller86** [comment]
We should relegate all the text we currently have in the Curated Content issue template to some markdown file here in the repo and just include a link to it from the CC template. A challenge here is that in order for that link to be click-able, the submitter has to know to go look at the issue via the *Preview* tab.  Would be nice if the template button just dropped a submitter into the issue on the *Preview* tab instead of the *Write* tab.

**Comment by JeanShuler** [comment]
I think write access should be restricted to a group of knowledgeable subject matter people/experts who could determine the relevancy of the content. 
I like the idea of doing a "CI" check for the PRs. 
Automation is great but I think there should be an option to do manual labeling.

**Comment by markcmiller86** [comment]
> My thought was that having multiple project boards would be useful to help us all visualize and follow the different processes for the different types of content. Also, the project boards support some automations which may (or may not) be useful to us. I don’t think you can do the same on bare issues.

Those are good reasons to use the Kanban/project boards. I hadn't considered that.

**Comment by markcmiller86** [comment]
> I think write access should be restricted to a group of knowledgeable subject matter people/experts who could determine the relevancy of the content.

@JeanShuler ... not sure if you are referring to my comment regarding *Write* vs. *Preview* tab but if you are...everyone needs ability to *write* an issue ;) So, we cannot get away from that. The point I was making is that in GitHub issue submission, the template text is not properly rendered markdown (with active links) and so if that text includes URLs to instruction pages somewhere else, one cannot actually follow those links from the *Write* tab. They have to go to the *Preview* tab.

**Comment by markcmiller86** [comment]
> It seemed to me that looking for an excess of positive votes was feasible from an automation standpoint and hopefully, more or less equivalent to what we discussed.

@bernhold It looks like a search filter like `is:issue is:open sort:reactions-+1-desc` can tease out issues with large vote counts.  Note that you can effectively *save* a given search filter you create by simple saving the URL for the page of results (which contains the encoded filter parameters)

**Comment by markcmiller86** [comment]
I think I've convinced myself that the check-boxes in the issue template to capture up-votes for the curated content submission are nowhere near as good as just using thumbs-up/+1 reactions. So, I suggest we agree to remove those from the issue template text also.

**Comment by markcmiller86** [comment]
From @bernhold 

> FYI, another way people can provide input about BSSw is to use the Contact form.  We got one today that has a suggestion for a curated content article.
 
> Lois and I discussed this a little bit, and my feeling is that we’re too early in our exploration of the editorial workflow to try to subject external people, friendly as they may be, to being used as guinea pigs.

**Comment by jarrah42** [comment]
Would it be valuable to separate the editorial board role from a reviewer role? Initially they might be the same, but if we want to expand participation to a broader community, it might be useful.

**Comment by jarrah42** [comment]
I think it would be good for an Editorial Board member to take ownership of the request to ensure that it reaches a conclusion. This was a characteristic of many of the workflows I reviewed.

**Comment by pagrubel** [comment]
I agree with Greg. At some point it would be nice if atleast the editorial board member who has ownership has a time frame in which to make sure the content is moving along.

**Comment by pagrubel** [comment]
I think the Question on Actors "Do we need to do anything..." should be replaced with either the link to the form where they can submit content ideas or a comment about it"

**Comment by markcmiller86** [comment]
@jarrah42 and @pagrubel see revisions as of c58ff18

File: Site/CuratedContentEditorialWorkflow.md [hunk 3]
```diff
@@ -52,3 +57,19 @@ Reference:
...
+   [GitHub web api](https://developer.github.com/v3/issues/milestones/#create-a-milestone). This
+   would give EB members one month granularity to set deadlines yet avoid creating a milestone each time
+   we wanna set a deadline.
+1. Would it be valuable to separate the editorial board role from a reviewer role?
+   Initially they might be the same, but if we want to expand participation to a broader
+   community, it might be useful.
+1. **Authors with no GitHub account:**
+   * Do we need to do anything to accommodate those who can’t (or don’t want to) submit a GH issue?
+     * [@pagrubel] provide a link to the form where they can submit content ideas or a comment about it"
+     * [@markcmiller86] No, lets not support this in any way other than via GH issue.
```
Comment (bartlettroscoe) [inline Site/CuratedContentEditorialWorkflow.md]: That is my vote.  GitHub is the process.

**Comment by bernhold** [comment]
> I think write access should be restricted to a group of knowledgeable subject matter people/experts who could determine the relevancy of the content.

I think to ensure the control we need that we will have to limit write access to the repository.  Probably to the Editorial Board members and Editorial Assistant(s).  Everyone else uses PRs.

**Comment by bernhold** [comment]
> Would it be valuable to separate the editorial board role from a reviewer role? Initially they might be the same, but if we want to expand participation to a broader community, it might be useful.

I have no problem with adding a role like that. But I think we should be thoughtful about how often we use them.  IMHO, the bar for curated content and events should not be terribly high.  Consequently, I assumed that the Editorial Board should be able to handle the reviews.  Another term to use for the role might be Subject Matter Expert.

**Comment by rinkug** [comment]
#5 An interested Author picks an issue from the Development Backlog column and produces a draft of the item as a pull request (PR), associated with the issue. The pull request is added to the Item Review column.

We need to have documentation on how to create a PR. Also, how is the author notified in #5 since they would not be scanning the development backlog column (will they even have permission to see it?)

**Comment by rinkug** [comment]
> #5 An interested Author picks an issue from the Development Backlog column and produces a draft of the item as a pull request (PR), associated with the issue. The pull request is added to the Item Review column.
> 
> We need to have documentation on how to create a PR. Also, how is the author notified in #5 since they would not be scanning the development backlog column (will they even have permission to see it?)

Alternatively there are several good and easy you tube videos that show to fork directory, modify and create a PR

**Comment by markcmiller86** [comment]
> We need to have documentation on how to create a PR.

Hmm... @rinkug do you mean having something that is somehow BSSw- specific? If so, how would it be different from just a normal PR? Not sure I agree we need anything here. I mean, there is already plenty out there. Maybe just a link to a decent article already available would be fine.

> Also, how is the author notified in #5 since they would not be scanning the development backlog column (will they even have permission to see it?)

@rinkug do you mean..."how is the *submitter* notified"? We can add a step to tell the Author to notify the submitter.

**Comment by rinkug** [comment]
> > We need to have documentation on how to create a PR.
> 
> Hmm... @rinkug do you mean having something that is somehow BSSw- specific? If so, how would it be different from just a normal PR? Not sure I agree we need anything here. I mean, there is already plenty out there. Maybe just a link to a decent article already available would be fine.

Yes, a link to an article on you tube video is fine. For ex: https://www.youtube.com/watch?v=e3bjQX9jIBk&t=157s

> > Also, how is the author notified in #5 since they would not be scanning the development backlog column (will they even have permission to see it?)
> 
> @rinkug do you mean..."how is the _submitter_ notified"? We can add a step to tell the Author to notify the submitter.

Yes, I meant the submitter.

**Comment by markcmiller86** [comment]
@rinkug ... I updated the [markdown file](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/blob/55236a5a8af1054ce0c913deff62f85077f8875e/Site/CuratedContentEditorialWorkflow.md) with information you suggested.

**Comment by bernhold** [comment]
I guess I assumed that the Editorial Board would be defined separately, so that in the context of this workflow, it would be a given.

**Comment by markcmiller86** [comment]
> I think it's fine for an initial version. Is there any reason that Editorial Board Member is not defined? Or is it assumed we know who this is?

For sure it needs to be defined *somewhere*. But, I don't think the right place is a document defining the *workflow* itself because the workflow doesn't or shouldn't change just because staffing changes. So, we use *symbolic constants* here in the workflow for actual people and somewhere else, we maintain the list of the *values* of those constants are currently set to as in

### Current Editorial Board Members
- @bernhold 
- @curfman

**Comment by markcmiller86** [comment]
@jarrah42 see most recent note (at bottom) added to document

**Comment by jarrah42** [comment]
> For sure it needs to be defined _somewhere_. But, I don't think the right place is a document defining the _workflow_ itself because the workflow doesn't or shouldn't change just because staffing changes. So, we use _symbolic constants_ here in the workflow for actual people and somewhere else, we maintain the list of the _values_ of those constants are currently set to as in

I wasn't suggesting that we list the members here, just add a definition of what the EB is, like the other definitions for Actors.

File: Site/CuratedContentEditorialWorkflow.md (hunk 1 above)
Comment (JeanShuler) [reply Site/CuratedContentEditorialWorkflow.md]: I agree that we should only have curated conted. _

**Pull request merged**